pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -242,8 +243,13 @@ func PodEligibleToPreemptOthers(pod *v1.Pod, nodeInfos framework.NodeInfoLister,
 			podPriority := corev1helpers.PodPriority(pod)
 			for _, p := range nodeInfo.Pods {
 				if p.Pod.DeletionTimestamp != nil && corev1helpers.PodPriority(p.Pod) < podPriority {
-					// There is a terminating pod on the nominated node.
-					return false
+					if p.Pod.Spec.TerminationGracePeriodSeconds != nil {
```

Comment:
What if there is no grace period?
We seek to preempt other nodes already?

I feel like this is too tight. We are not giving any time for the kubelet to remove the Pod and that to propagate to the scheduler. Same if there is a grace period

Reply:
> What if there is no grace period?
this won't happen,  please see,  https://github.com/kubernetes/kubernetes/blob/ff4234720d4ec390ca6f6aac4c19a21ba6661841/staging/src/k8s.io/api/core/v1/types.go#L2965-L2967

Reply:
> I feel like this is too tight. We are not giving any time for the kubelet to remove the Pod and that to propagate to the scheduler. Same if there is a grace period

I will extend the time to add a toleration period which is defined in scheduler.